### ISSUE TITLE
Consider adding setting that only shows inlay hint parameter names for simple literals (string, number, bool)

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#45295]
# Suggestion

## 🔍 Search Terms
- inlay hints
- parameter names
- typescript.inlayHints.parameterNames.enabled

## ✅ Viability Checklist

N/A, tooling related

## ⭐ Suggestion

Add a new setting that only shows inlay hints for simple literals. This would include:

- string
- number
- bool
- regexp literal?
- null?
- undefined?

Currently the `"typescript.inlayHints.parameterNames.enabled": "literals"` setting also shows parameter name hints for function, array, and object literals

I imagine this new setting would be called something like:  `"typescript.inlayHints.parameterNames.enabled": "simpleLiterals"`, although better names welcome

**Comment by a-tarasyuk** [comment]
@mjbvz Is it possible to have an array as an option? To allow users to choose what literals should be handled. Or adding `simpleLiterals` option to handle `"string", "number", "bool"` would be enough?

```ts
"literals" | ["string", "number", "bool", "regexp", "null", "undefined"]
```

**Comment by mjbvz** [comment]
Yes, we just would need to make sure VS Code's settings editor can render it nicely

One option is to have settings per type: `typescript.inlayHints.paramterNames.showStings`, `typescript.inlayHints.paramterNames.showNumber`, ....

This is what we do for the suggest settings in VS Code, such as `editor.suggest.showEnumMembers`


The other option would be to have a single setting with checkboxes. This is similar to what the gitLens extension does:

![Screen Shot 2021-08-03 at 2 25 42 PM](https://user-images.githubusercontent.com/12821956/128088435-4ed19c66-9136-47b5-aa9f-cc3af88465fc.png)

---

On the TS side, we could provide the setting in whatever format is most convenient for you

On the VS Code side, I think I'd prefer the first approach for discoverability. This is getting a bit beyond the original issue's scope, but the main options I'd personally want are:

- `objects`
- `functions`
- `simple literals`: `string`, `number`, `bool`, `regexp`, `undefined`, `null`

**Comment by a-tarasyuk** [comment]
@mjbvz Thanks for the detailed clarification. 
 
Currently option`"literals"` handles the following expressions/literals
```
string, number, bool, regexp, undefined, null, [], {}, () => {}, function () {}, ``
```

New options:

`"literals"` - Current behavior shouldn't be changed
`"simple literals"` - `string`, `number`, `bool`, `regexp`, `undefined`, `null`
`"functions"` - `() => {}`, `function () {}`
`"objects"`  - `{}` 

Which option should handle `[]`?

**Comment by jessetrinity** [comment]
>`"simple literals" - string, number, bool, regexp, undefined, null`

Suggestion:
`"primitive literals" - string, number, bool, undefined, null`

**Comment by a-tarasyuk** [comment]
@mjbvz I can try to create a POC with one option `"primitiveLiterals"` which will handle only `string`, `number`, `bool`, `undefined`, `null`. The option `"literals"` will work without changes. Does that make sense?

**Comment by mjbvz** [comment]
Yes that sounds great. I think that would become the new default for VS Code
